model/src/form/form-definition/index.ts

```diff
@@ -316,6 +324,42 @@ export const conditionDataSchemaV2 = Joi.object<ConditionDataV2>()
         'Value to compare the field against, either fixed or relative date'
       )
   })
+  .custom((value: ConditionDataV2, helpers: CustomHelpers<ConditionDataV2>) => {
+    const { componentId } = value
+    const definition = helpers.state.ancestors.find(
+      (item: FormDefinition) =>
+        'name' in item && 'pages' in item && 'conditions' in item
+    )
+
+    // Validation may not have been fired on the full FormDefinition
+    // therefore we are unable to verify at this point, but the 'save'
+    // will eventually validate the full FormDefinition
+    if (!isFormDefinition(definition)) {
+      return value
+    }
+
+    const foundComponents = definition.pages
+      .map((page) =>
+        hasComponentsEvenIfNoNext(page)
+          ? page.components.find((comp) => comp.id === componentId)
+          : undefined
+      )
+      .filter(Boolean)
+
+    const foundComponentHandlesConditions = foundComponents.length
+      ? isConditionalType(foundComponents[0]?.type)
+      : false
+
+    return foundComponentHandlesConditions
+      ? value
+      : helpers.error('any.ref', {
```

Comment:
We probably shouldn't use `'any.ref'` as the error code. Maybe `helpers.error('custom.incompatible',`?
Also, do `arg` and `ref` align with other `joi` local context props? I don't remember seeing them ever for other errors but others do have `key` and `path`